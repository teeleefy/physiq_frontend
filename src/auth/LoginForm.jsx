import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import './styles/Login.css'
import { useState } from 'react'
import { useNavigate } from "react-router-dom";
import { toast } from 'react-toastify';

function LoginForm({login}){
    const INITIAL_STATE = { email: "", password: ""};
    const [formData, setFormData] = useState(INITIAL_STATE);
    const [formErrors, setFormErrors] = useState([]);
    const navigate = useNavigate();
    /** Send { name, description, recipe, serve } to parent
     *    & clear form. */
  
    async function handleSubmit(evt){
        evt.preventDefault();
        // console.log(formData);
        let result = await login(formData);
        if(result.success){
            toast.success('Logged In!', {
                position: "top-center",
                autoClose: 2000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
                theme: "dark",
                });
            setFormData(INITIAL_STATE);
            navigate('/');
        } else{
            setFormErrors(result.errors);
        }
    };
  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };

    return(
        <>
            
            <Form className="Login-Form" onSubmit={handleSubmit}>
            <h1 className="Login-h1">Login</h1>
            <FormGroup>
                <Label for="email" >
                    <b>Email</b>
                </Label>
                <Input
                id="email"
                name="email"
                placeholder="Enter email..."
                minLength={6}
                maxLength={60}
                type="text"
                value={formData.email}
                onChange={ handleChange }
                />
            </FormGroup>
            <FormGroup>
                <Label for="password">
                    <b>Password</b>
                </Label>
                <Input
                id="password"
                name="password"
                placeholder="Enter password..."
                value={formData.password}
                type="password"
                minLength={8}
                onChange={ handleChange }
                />
                </FormGroup>

                <Button className="Login-btn btn-dark">Submit</Button>
                {formErrors.length
                    ? formErrors.map(error => <Alert color="danger">{error}</Alert>)
                    : null
                }

                
            </Form>
        </>
     )
}

export default LoginForm;