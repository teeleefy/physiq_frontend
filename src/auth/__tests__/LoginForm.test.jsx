import { renderWithRouter, screen } from '../../../__tests__/test-tools';
import LoginForm from '../LoginForm'

describe('LoginForm', () => {
  it('renders the LoginForm component', () => {
    renderWithRouter(<LoginForm />)
    
    screen.debug(); // prints out the jsx in the LoginForm component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<LoginForm />);
  expect(asFragment()).toMatchSnapshot();
});