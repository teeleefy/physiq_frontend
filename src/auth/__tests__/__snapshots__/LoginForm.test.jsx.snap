// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Login-Form"
  >
    <h1
      class="Login-h1"
    >
      Login
    </h1>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="email"
      >
        <b>
          Email
        </b>
      </label>
      <input
        class="form-control"
        id="email"
        maxlength="60"
        minlength="6"
        name="email"
        placeholder="Enter email..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="password"
      >
        <b>
          Password
        </b>
      </label>
      <input
        class="form-control"
        id="password"
        minlength="8"
        name="password"
        placeholder="Enter password..."
        type="password"
        value=""
      />
    </div>
    <button
      class="Login-btn btn-dark btn btn-secondary"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
